pageserver/src/walingest.rs

```diff
@@ -104,12 +103,66 @@ struct WarnIngestLag {
     timestamp_invalid_msg_ratelimit: RateLimit,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum WalIngestError {
+    #[error(transparent)]
+    #[allow(private_interfaces)]
+    PageReconstructError(#[from] PageReconstructError),
+    #[error(transparent)]
+    DeserializationFailure(#[from] DeserializeError),
+    #[error(transparent)]
+    SerializationFailure(#[from] SerializeError),
+    #[error("the request contains data not supported by pageserver: {0} @ {1}")]
+    InvalidKey(Key, Lsn),
+    #[error("twophase file for xid {0} already exists")]
+    FileAlreadyExists(u64),
+    #[error("slru segment {0:?}/{1} already exists")]
+    SlruAlreadyExists(SlruKind, u32),
+    #[error("relation already exists")]
+    RelationAlreadyExists,
+    #[error("invalid relnode")]
+    InvalidRelnode,
+    #[error("invalid reldir key")]
+    InvalidRelDirKey,
+
+    #[error(transparent)]
+    AssertionError(anyhow::Error),
```

Comment:
How about renaming to `LogicalError`. It isn't an assertion in the real sense.

Reply:
done